sdk/storage/src/queue/requests/delete_message_builder.rs

```diff
@@ -0,0 +1,167 @@
+    timeout: Option<u64>,
+    client_request_id: Option<&'a str>,
+}
+
+impl<'a, C> DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    #[inline]
+    pub(crate) fn new(
+        queue_name_service: &'a dyn QueueNameService<StorageClient = C>,
+    ) -> DeleteMessageBuilder<'a, C> {
+        DeleteMessageBuilder {
+            queue_name_service,
+            pop_receipt: None,
+            timeout: None,
+            client_request_id: None,
+        }
+    }
+}
+
+//set mandatory no traits methods
+impl<'a, C> PopReceiptRequired for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    #[inline]
+    fn pop_receipt(&self) -> &dyn PopReceipt {
+        self.pop_receipt.as_deref().unwrap()
+    }
+}
+
+impl<'a, C> TimeoutOption for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    #[inline]
+    fn timeout(&self) -> Option<u64> {
+        self.timeout
+    }
+}
+
+impl<'a, C> ClientRequestIdOption<'a> for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    #[inline]
+    fn client_request_id(&self) -> Option<&'a str> {
+        self.client_request_id
+    }
+}
+
+impl<'a, C> PopReceiptSupport for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    type O = DeleteMessageBuilder<'a, C>;
+
+    #[inline]
+    fn with_pop_receipt(self, pop_receipt: Box<dyn PopReceipt>) -> Self::O {
+        DeleteMessageBuilder {
+            queue_name_service: self.queue_name_service,
+            pop_receipt: Some(pop_receipt),
+            timeout: self.timeout,
+            client_request_id: self.client_request_id,
+        }
+    }
+}
+
+impl<'a, C> TimeoutSupport for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    type O = DeleteMessageBuilder<'a, C>;
+
+    #[inline]
+    fn with_timeout(self, timeout: u64) -> Self::O {
+        DeleteMessageBuilder {
+            queue_name_service: self.queue_name_service,
+            pop_receipt: self.pop_receipt,
+            timeout: Some(timeout),
+            client_request_id: self.client_request_id,
+        }
+    }
+}
+
+impl<'a, C> ClientRequestIdSupport<'a> for DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    type O = DeleteMessageBuilder<'a, C>;
+
+    #[inline]
+    fn with_client_request_id(self, client_request_id: &'a str) -> Self::O {
+        DeleteMessageBuilder {
+            queue_name_service: self.queue_name_service,
+            pop_receipt: self.pop_receipt,
+            timeout: self.timeout,
+            client_request_id: Some(client_request_id),
+        }
+    }
+}
+
+// methods callable regardless
+impl<'a, C> DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    pub fn queue_name_service(&self) -> &'a dyn QueueNameService<StorageClient = C> {
+        self.queue_name_service
+    }
+}
+
+// methods callable only when every mandatory field has been filled
+impl<'a, C> DeleteMessageBuilder<'a, C>
+where
+    C: Client,
+{
+    pub async fn execute(self) -> Result<DeleteMessageResponse, AzureError> {
```

Comment:
It seems it would be possibile to call `execute` without having set the mandatory `pop_receipt`. This will cause a runtime panic (the `unwrap` on the getter). 

We can fix this after this one is merged.

Reply:
Ahh right, that'll be all the `Yes` and `No` stuff in `PutMessageBuilder` right? I was wondering what that did.

Reply:
Yes, I follow this approach for compile-time enforced builders: [https://dev.to/mindflavor/rust-builder-pattern-with-types-3chf](https://dev.to/mindflavor/rust-builder-pattern-with-types-3chf).

This is nice for the user perspective but adds a lot of boilerplate code to the crate maintainers.